app/models/appdata.rb

```diff
@@ -34,7 +36,10 @@ def self.add_appdata data, xml
   # Get the appdata xml for a distribution
   def self.get_distribution dist="factory", flavour="oss"
     if dist == "factory"
-      appdata_url = "http://download.opensuse.org/tumbleweed/repo/#{flavour}/suse/setup/descr/appdata.xml.gz"
+      appdata_regex = /appdata-\S{32}.xml.gz/
+      tumbleweed_contents = open('http://download.opensuse.org/tumbleweed/repo/oss/content') {|f| f.read }
```

Comment:
Should probably be ttp://download.opensuse.org/tumbleweed/repo/#{flavour}/content - to not limit it to the oss repo